### PR TITLE
feat: add mvi list indexes detail

### PR DESCRIPTION
**Pull request by malandis**
Add MVI list indexes detail to IndexInfo object and update integration tests.

### CODE REVIEW

File: src/Momento.Sdk/Responses/Vector/IndexInfo.cs [hunk 1]
```diff
@@ -10,30 +12,52 @@ public class IndexInfo
...
     /// <inheritdoc />
     public override int GetHashCode()
     {
-        return Name.GetHashCode();
+        unchecked
+        {
+            var hash = 17;
+            hash = hash * 23 + Name.GetHashCode();
+            hash = hash * 23 + NumDimensions.GetHashCode();
+            hash = hash * 23 + SimilarityMetric.GetHashCode();
```
Comment (eaddingtonwhite) [inline src/Momento.Sdk/Responses/Vector/IndexInfo.cs:53]: JC where do 17 and 23 come from? Could we make named constants for them for what they represent? Just not clear to me looking at code why they equal to what they are.

File: src/Momento.Sdk/Responses/Vector/IndexInfo.cs [hunk 2]
```diff
@@ -10,30 +12,52 @@ public class IndexInfo
...
     {
-        return obj is IndexInfo other && Name == other.Name;
+        return obj is IndexInfo other && Name == other.Name && NumDimensions == other.NumDimensions && SimilarityMetric == other.SimilarityMetric;
     }
 
     /// <inheritdoc />
     public override int GetHashCode()
     {
-        return Name.GetHashCode();
+        unchecked
```
Comment (eaddingtonwhite) [inline src/Momento.Sdk/Responses/Vector/IndexInfo.cs:48]: this seems scary to me seeing this. Maybe just leave comment why we need?

**Review by eaddingtonwhite — COMMENTED** [review]
@malandis overall looks good. Just 2 nits around adding some context for next dev that comes in here in comments.

File: src/Momento.Sdk/Responses/Vector/IndexInfo.cs (hunk 1 above)
Comment (malandis) [reply src/Momento.Sdk/Responses/Vector/IndexInfo.cs:53]: These are prime numbers used to combine the hash codes. There is a .NET standard implementation to combine the hash code representations, but we are targeting such old frameworks we don't have access to it, hence have to copy that here. See here https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/HashCode.cs#L58

Comment (eaddingtonwhite) [reply src/Momento.Sdk/Responses/Vector/IndexInfo.cs:53]: I see ok maybe just comment in code linking out to this at least? I would never have figured this out on my own or had context why were doing.

**Comment by malandis** [comment]
> @malandis overall looks good. Just 2 nits around adding some context for next dev that comes in here in comments.

@eaddingtonwhite done

**Review by eaddingtonwhite — APPROVED** [review]

**Pull request merged**